Jenkinsfile

```diff
@@ -75,7 +101,163 @@ void createPipelineTriggers() {
     }
 }
 
-def step_compile_results()
+/* Gets the robot framework url and commit id based on pipeline parameters.
+ *
+ * Sets rfUrl and rfCommitId.
+ */
+def stepGetRfUrlAndCommit() {
+    (rfUrl, rfCommitId) = getUrlAndCommitFromParams(
+            params.HIL_RF_VERSION, "RIOT-OS",
+            "RobotFW-Tests", params.HIL_RF_PULL)
+    sh script: "echo rfUrl ${rfUrl}", label: "rfUrl ${rfUrl}"
+    sh script: "echo rfCommitId ${rfCommitId}",
+            label: "rfUrl ${rfCommitId}"
+}
+
+/* Gets the riot url and commit id based on pipeline parameters.
+ *
+ * Sets riotUrl and riotCommitId.
+ */
+def stepGetRiotUrlAndCommit() {
+    (riotUrl, riotCommitId) = getUrlAndCommitFromParams(
+            params.HIL_RIOT_VERSION, "RIOT-OS",
+            "RIOT", params.HIL_RIOT_PULL)
+    sh script: "echo riotUrl ${riotUrl}", label: "riotUrl ${riotUrl}"
+    sh script: "echo riotCommitId ${riotCommitId}",
+            label: "riotCommitId ${riotCommitId}"
+}
+
+/* Gets github url and commit based on parameters.
+ *
+ * @param ver           Type of version to get {master, pull, current, submodule}
+ * @param repo_owner    Repoisitory owner
+ * @param repo_name     Repoisitory name
+ * @param pull          Optional pull request number
+ *
+ * @return (url, commit)
+ */
+def getUrlAndCommitFromParams(ver, repo_owner, repo_name, pull = 0) {
+    if (ver == 'master') {
+        return getUrlAndCommitFromDefault(repo_owner, repo_name)
+    }
+    else if (ver == 'pull') {
+        assert pull != '0' : "Selected pull of ${repo_name} but no pull entered"
+        return getUrlAndCommitFromPr(repo_owner, repo_name, pull)
+    }
+    else if (ver == 'current') {
+        return ["https://github.com/${repo_owner}/${repo_name}",
+                env.GIT_COMMIT]
+    }
+    else if (ver == 'submodule') {
+        return getUrlAndCommitOfSubmodule("RIOT-OS", "RobotFW-Tests")
+    }
+    assert false : "No valid options to get commit and url of ${repo_name}"
+}
+
+/* Gets the SHA1 commit id and url of a github default branch.
+ *
+ * @param repo_owner    The owner of the repo.
+ * @param repo_name     The name of the repo.
+ *
+ * @return              (url, commit)
+ */
+def getUrlAndCommitFromDefault(repo_owner, repo_name) {
+    def query = "-X POST -d \"{\\\"query\\\": \\\"query "
+    query = "${query}{repository(name: \\\\\\\"${repo_name}\\\\\\\","
+    query = "${query}owner: \\\\\\\"${repo_owner}\\\\\\\") "
+    query = "${query}{defaultBranchRef "
+    query = "${query}{ target { oid } }}}\\\"}\" "
+    query = "${query}https://api.github.com/graphql"
+    def res = ""
+    withCredentials([string(credentialsId: 'mrkevinweiss_github_api_token',
+            variable: 'TOKEN')]) {
+        res = sh script: '''
+        curl -H "Authorization: token $TOKEN" ''' + query,
+        label: "Fetching commit id", returnStdout: true
+    }
```

Comment:
this is used multiple times with different queries. maybe refactor it to a function accepting the query as the parameter?